docs/data-formats.md

```diff
@@ -4,7 +4,23 @@ IL linker uses several data formats to control or influence the linking process.
 
 ## Descriptor Format
 
-The `fullname` attribute specifies the fullname of the type in the format specified by ECMA-335. This is in certain cases not the same as the one reported by Type.FullName for example for nested types.
+Descriptors are used to direct linker to always keep some items in the assembly, regardless if linker can find any references to them.
+
+Descriptor XML can be embedded in an assembly. In that case it must be stored as an embedded resource with logical name `ILLink.Descriptors.xml`. To achieve this when build an assembly use this in the project file to include the XML:
+
+```xml
+  <ItemGroup>
+    <EmbeddedResource Include="ILLink.Descriptors.xml">
+      <LogicalName>ILLink.Descriptors.xml</LogicalName>
+    </EmbeddedResource>
+  </ItemGroup>
+```
+
+Embedded descriptors only take effect if the containing assembly is included in the linker output, so if something from that assembly is marked to be kept.
+
+Descriptor XML can also be passed to the linker on the command via the [`-x` parameter](https://github.com/mono/linker/blob/master/docs/illink-options.md#linking-from-an-xml-descriptor).
+
+The `fullname` attribute specifies the full name of the type in the format specified by ECMA-335. This is in certain cases not the same as the one reported by Type.FullName for example for nested types.
```

Comment:
While you're touching this, we should delete this description of fullname. The ECMA-335 format this refers to is https://github.com/dotnet/runtime/issues/4416, but what linker uses is not that format.

Fullname in linker is the full name as understood by Cecil, period. (Cecil differs from ECMA in generics, it differs by the fact that Cecil's fullname cannot distinguish `szArray` and `singleDimensionalArray` in the ECMA doc linked above, it differs in escaping rules because Cecil has none, etc.).

Reply:
I simply removed that line - it's basically useless anyway - if we want better, we should add a list of examples instead.